### DISCUSSION TITLE
Is there a plan for the Composable Architecture if things go south?

### DISCUSSION THREAD

**Discussion by stremsdoerfer** [pointfreeco/swift-composable-architecture#2000] [General]
Hello,

We're building an app in a professional environment, we love the Composable Architecture but the main concern that we have is that it is a 3rd party library.

Like with any 3rd party library, there's a bit of a gamble. In most cases, we can abstract things enough that if the library was to disappear we could replace it with another one or a custom implementation pretty easily.

The Composable Architecture is a little different though as it defines foundations for big parts of the app. It would be a significant loss for us if things were to go south for any reason let's say in the next 5 years.

We're wondering if there's a plan for that and if it's a concern you'd like to address?

**Comment by mbrandonw** [comment]
Hi @stremsdoerfer, we feel we have done quite a bit of work to make TCA a somewhat safe dependency to take on. The community around TCA is vibrant, there is a lot of great documentation from us and the community, and of course there is an entire video series that describes nearly every choice we made and why we made it. In fact, I'm not really aware of any other library out there that has such a large source of knowledge of why the library is the way it is.

However, at the end of the day, bringing a 3rd party dependency into a code base is a big responsibility, and you should play an active role. You should probably subscribe to this repo's every update (as well as the other libraries we use) so that you can be up-to-date with each change, you should participate in the community, and you should be deeply familiar with the internals of the library.

If you do that then it can be less scary taking on any dependency, not just TCA. And of course our plan is to continue being the shepherds of TCA and all of our other open source projects. We are lucky that this effort is funded by Point-Free subscriptions, and we'd love if any companies using our libraries subscribe in order to support future development.

And if people have some concrete suggestions of things to explore in the future we are of course all ears. :)

**Comment by stremsdoerfer** [comment]
Thanks for your quick response!
You bring up good points and I definitely feel like a lot has been done to alleviate that and I really appreciate it.

- Vibrant community. 100% agree. There's a lot of interest and passionate people around TCA. I've since realized that there was a [Slack](http://pointfree.co/slack-invite) that is pretty active and it's nice to see. 
- Thorough documentation indeed. Overall probably better than any team i've worked on could have built, and because of that, I believe it's not bad to have a 3rd party dependency, rather than a custom in-house framework that is poorly documented. 

As for concrete suggestions, I think TCA could get more attention and therefor more contributors to improve the "bus factor". A simple search for [iOS SwiftUI architecture](https://www.google.com/search?q=ios+swiftui+architectures&oq=ios+swiftui+architectures) yields a bit outdated results and TCA is barely present despite being one of the most used amongst more advanced developers. I think Medium articles could help here, and i'd be happy to participate in there.
